### DISCUSSION TITLE
UnlockAchievements UnexpectedError not handled

### DISCUSSION THREAD

**Discussion by rekterakathom** [FakeFishGames/Barotrauma#14434] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Multiple messages ``Result "UnexpectedError" was not handled by "UnlockAchievements"`` in console. Seemingly after every mission completed. Some people in the lobby unlocked achievements during the session.

![20240814214430_1](https://github.com/user-attachments/assets/fbc19da2-a5e8-4def-9632-c71404cd1d25)

### Reproduction steps

1. Finish a mission (during a session where somebody unlocked an achievement...?)
2. Open console
3. Check for UnexpectedError messages

### Bug prevalence

Just once

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly. Fix coming in the next patch after the upcoming hotfix update. Closing discussion.

**Reply by Randy624** [reply]
Have the same error in may 2025.
